File: 08_nativeModule/nativeConnectionSetup/ios/MyNativeModule.m
//
//  MyNativeModule.m
//  nativeConnectionSetup
//
//  Created by Chaitanya on 30/03/26.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>


@interface RCT_EXTERN_MODULE(MyNativeModule, NSObject)
RCT_EXTERN_METHOD(showMessage: (NSString *) message)
RCT_EXTERN_METHOD(openNewScreen)
RCT_EXTERN_METHOD(requiresMainQueueSetup)

@end

File: 08_nativeModule/nativeConnectionSetup/ios/MyNativeModule.swift
//
//  MyNativeModule.swift
//  nativeConnectionSetup
//
//  Created by Chaitanya on 30/03/26.
//

import Foundation
import React
import UIKit

@objc(MyNativeModule)
class MyNativeModule: NSObject {

    @objc
    func showMessage(_ message: String) {
        print(message)
    }

    @objc
    func openNewScreen() {
        DispatchQueue.main.async {
            if let rootVC = UIApplication.shared.keyWindow?.rootViewController {
                let newVC = NewScreenViewController()
                rootVC.present(newVC, animated: true, completion: nil)
            }
        }
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
